[Anselm Pius Long] iP

anselmlong opened the pull request:
## Sigma
> Not only do I survive but thrive, for I am born to be a Sigma male. - George Washington [^1]

Want a chatbot to keep track of your tasks, but everything else is too beta for you?
If you're a true **Sigma**, you need to:
- Be on top of your tasks
- Be on the grind set
- ~~Give Up~~ Get back up when you fail
> Along the way, I will stumble a hundred times, but I will get back up to my feet a hundred and more. - Someone important, probably 

Ready to be a *real Sigma?*
1. Download the `sigma.jar` file from [here](https://github.com/anselmlong/ip/releases/tag/v0.2).
2. Open it and follow the instructions inside.
3. Be ready to have your life transformed.

***What the sigma?*** 🧍‍♂️⁉️💣  

What can you do with Sigma?
- [X] Add tasks
- [X] Manage tasks
- [x] Get Reminded (coming soon!)
- [x] Get a motivational quote (coming soon!)
- [ ] Get a girlfriend (hopefully!)

Want to see what Sigma has in store for you? Here's an example:
``` java
public static void exit() {
        print("What the sigma? You're leaving so soon? Bye chat, see you again!");
}
```

> A motivated man is strong, but a disciplined man is deadly. - Donald Trump [^2]

[^1]: May be exaggerated.
[^2]: He may or may not have said this.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

anselmlong committed:
Rename project and added exit message

anselmlong committed:
Add echo feature

anselmlong committed:
Add list feature

anselmlong committed:
Add task class, mark and unmark feature

anselmlong committed:
Add different task types

anselmlong committed:
Add test

anselmlong committed:
Add error handling with SigmaExceptions

anselmlong committed:
Add delete feature

anselmlong committed:
Add enums

anselmlong committed:
New branch

anselmlong committed:
Commit feature1

anselmlong committed:
Add writing and reading from file

anselmlong committed:
Add writing and reading from file

anselmlong committed:
Add writing and reading to file

anselmlong committed:
Add proper date and time support

anselmlong committed:
Add more OOP classes

anselmlong committed:
Sort into packages

anselmlong committed:
Merge branch 'A-Packages'

anselmlong committed:
Merge branch 'add-gradle-support'

anselmlong committed:
Add Gradle

anselmlong committed:
Merge branch 'A-Gradle'

anselmlong committed:
Add unit tests for some command files

anselmlong committed:
Merging to master branch

anselmlong committed:
Add Javadoc for all classes

anselmlong committed:
Modify code to follow coding standard

anselmlong committed:
Add find functionality

anselmlong committed:
Merge JavaDoc

anselmlong committed:
Fix merge conflicts

anselmlong committed:
Fix merge conflicts for Level 9

seanlim commented on build.gradle:
```diff
@@ -0,0 +1,42 @@
+}
+
+application {
+    mainClass.set("sigma.Sigma")
```
Nice lol

seanlim commented on data.txt:
Hmm actually is there any reason why you checked this file in? You can ignore it by adding it to your `.gitignore`

anselmlong replied:
Oh, I didn't know that. Thanks!

seanlim commented on src/main/java/sigma/Parser.java:
```diff
@@ -0,0 +1,73 @@
+    public static Commands parse(String command) {
+        String[] split = command.split(" ", 2);
+        String cmd = split[0]; // Convert input to uppercase to match enum
+        switch (cmd) {
```
Good use of `switch` here!

seanlim commented on src/main/java/sigma/Parser.java:
```diff
@@ -0,0 +1,73 @@
+     * @return LocalDateTime object.
+     * @throws SigmaException If the date and time input is invalid.
+     */
+    public static LocalDateTime dateTimeParse(String timing) throws SigmaException {
```
`parseLocalDateTime` might be a better name for this method for convention and clarity.

anselmlong replied:
Good catch. Changed!

seanlim commented on src/main/java/sigma/Storage.java:
```diff
@@ -0,0 +1,153 @@
+                                ? ((DeadlineTask) item).getDate()
+                                : item instanceof EventTask
+                                ? ((EventTask) item).getFrom() + " - " + ((EventTask) item).getTo()
+                                : ""));
```
I found this nested ternary operator very hard to read, it might be more easy to understand if you refactor this logic to earlier lines using `if-else` statements

anselmlong replied:
You're right, I'll change this!

anselmlong replied:
Changed to:
``` java
for (Task item : items) {
                String date;
                if (item instanceof DeadlineTask) {
                    date = ((DeadlineTask) item).getDate();
                } else if (item instanceof EventTask) {
                    date = ((EventTask) item).getFrom() + " - " + ((EventTask) item).getTo();
                } else {
                    date = "";
                }
                writer.write(String.format("%s | %s | %s | %s \n",
                        item.getTaskType(), item.getStatusString(), item.getDesc(), date));
            }
            writer.close();
```

seanlim commented on src/main/java/sigma/Storage.java:
```diff
@@ -0,0 +1,153 @@
+            LocalDateTime from;
+            LocalDateTime to;
+            try {
+                from = LocalDateTime.parse(dates[0].strip(), DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm"));
+                to = LocalDateTime.parse(dates[1].strip(), DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm"));
```
Perhaps declare `DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm")` as a constant? Then you don't need to change both lines if the date format changes in the future.

anselmlong replied:
Will do!

seanlim commented on src/main/java/sigma/Storage.java:
```diff
@@ -0,0 +1,153 @@
+        case "D":
+            LocalDateTime dateTime;
+            try {
+                dateTime = LocalDateTime.parse(date.strip(), DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm"));
```
See previous comment

seanlim commented on src/main/java/sigma/TaskList.java:
```diff
@@ -0,0 +1,54 @@
+ * Contains operations to add, remove, and get tasks.
+ * Also contains operations to check if the list is empty and to get the size of the list.
+ */
+public class TaskList {
```
Well-defined `TaskList` class! Just remember to add JavaDoc header comments

anselmlong replied:
Alright!

seanlim commented on src/main/java/sigma/Ui.java:
```diff
@@ -0,0 +1,60 @@
+ * Represents the user interface of Sigma.
+ * Contains methods to read user input and print messages.
+ */
+public class Ui {
```
Missing JavaDoc header comments, looks good otherwise!

seanlim commented on src/main/java/sigma/command/Commands.java:
```diff
@@ -0,0 +1,36 @@
+     * @param storage
+     * @throws SigmaException
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws SigmaException;
```
This violates the Java coding standard, gotta wrap into a newline. See https://se-education.org/guides/conventions/java/intermediate.html#layout

anselmlong replied:
Isn't it for lines more than 120 characters? I have about 95 here.

seanlim reviewed (commented):
Good Job!

Very entertaining chatbot indeed 😆. Just some minor coding standard issues, though I didn't highlight all of them because I believe you are probably aware of them yourself.

Also gave a few suggestions to improve code readability and overall make it easier to work with.

anselmlong commented:
> Good Job!
> 
> Very entertaining chatbot indeed 😆. Just some minor coding standard issues, though I didn't highlight all of them because I believe you are probably aware of them yourself.
> 
> Also gave a few suggestions to improve code readability and overall make it easier to work with.

Thank you bouldermonster69! Appreciate the comments and feedback :)

anselmlong committed:
Change naming, update coding standard in accordance with PR Review by sean

anselmlong committed:
Add GUI functionality for Sigma

Refactor commands to return String instead, to redirect output to GUI

anselmlong committed:
Fix merge with fast forward

anselmlong committed:
Merge tag 'Level-10'

DanzaSeah commented on src/main/java/sigma/task/ToDoTask.java:
```diff
@@ -0,0 +1,20 @@
+/**
+ * Represents a to-do task.
+ */
+public class ToDoTask extends Task {
```
May I give a suggestion, the class ToDoTask can be simplified to TodoTask, as the word ToDo is not camel case in most Java naming conventions, though it is still fine if you leave it as it is.

anselmlong replied:
Okay

DanzaSeah commented on src/main/java/sigma/utils/Storage.java:
```diff
@@ -0,0 +1,158 @@
+            FileWriter writer = new FileWriter(data);
+            for (Task item : items) {
+                String date;
+                if (item instanceof DeadlineTask) {
```
I noticed that you used of instanceof to dictate the different actions to perform depending on the class,  you can add an abstract method in the Task class that each subclass (e.g., DeadlineTask, EventTask) can override, this way you may save some lines of code on checking and casting.

anselmlong replied:
Oh right, good catch. Thanks!

DanzaSeah commented on src/main/java/sigma/command/ToDoCommand.java:
```diff
@@ -0,0 +1,41 @@
+ */
+public class ToDoCommand extends Command {
+
+    public ToDoCommand(String[] split) {
```
I noticed that you named the parameter split, may I suggest another name like commandArguments or inputParts which may better clarify its purpose?

DanzaSeah commented on src/main/java/sigma/command/ToDoCommand.java:
```diff
@@ -0,0 +1,41 @@
+     * @param ui
+     * @param storage
+     * @throws SigmaException
+     */
```
I noticed that there’s no blank line between the description and the @param tag in the Javadoc for this method and your other comments. It would be great if you would leave a blank line between the description and the parameter section for readability https://se-education.org/guides/conventions/java/intermediate.html#comments

anselmlong replied:
I see!

DanzaSeah commented on src/main/java/sigma/task/DeadlineTask.java:
```diff
@@ -0,0 +1,43 @@
+public class DeadlineTask extends Task {
+    private LocalDateTime date;
+
+    public DeadlineTask(String desc, LocalDateTime date) {
```
I noticed you used a short form word "desc", may I suggest that you used the full word description to completely avoid ambiguity?
``` suggestion
    public DeadlineTask(String description, LocalDateTime date) {

DanzaSeah commented on src/main/java/sigma/task/Task.java:
```diff
@@ -0,0 +1,44 @@
+ * Represents a task.
+ */
+public class Task {
+    private boolean status;
```
I noticed you named the boolean variable "status", may I suggest an alternative name "isCompleted" instead?
https://se-education.org/guides/conventions/java/intermediate.html#comments

anselmlong replied:
Good catch, thanks for the correction!

DanzaSeah reviewed (commented):
Great work on this project! I like the theme, you are truly a Sigma. The overall structure is well thought-out, and it's clear that a lot of effort went into building a solid foundation for the Sigma chatbot. The code is mostly easy to follow, and I appreciate the use of comments to describe the purpose of different sections, which helps with readability.

anselmlong committed:
Refine chatbot and redirect more outputs

anselmlong committed:
Refactor and improve GUI

anselmlong committed:
Comment out main and run function under Sigma

anselmlong committed:
Merge Conflict: Resolve from branch Level 7

Branch Level 7 was not merged properly.

As such, I did not have the correct status on the
dashboard. Now, I have merged it back into the
master branch, and resolved the merge conflicts.

This is done this way to ensure that everything
is up to date and clean.

In the future, I will regularly check for isolated
branches.

anselmlong committed:
Checkstyle: reformat project to follow checkstyle conventions

anselmlong committed:
Checkstyle: reformat to include checkstyle

anselmlong committed:
Merge branch 'branch-A-Checkstyle'

SeeYangZhi commented:
> Good Job!
> 
> 
> 
> Very entertaining chatbot indeed 😆. Just some minor coding standard issues, though I didn't highlight all of them because I believe you are probably aware of them yourself.
> 
> 
> 
> Also gave a few suggestions to improve code readability and overall make it easier to work with.

LGMT!

anselmlong committed:
Implement assertions in the codebase

The current codebase does not have assertions implemented.

This commit adds assertions to the codebase to improve the robustness
of the code and to help with debugging.

Assertions are added to check for null values and to ensure that the
code behaves as expected.

With this commit, the codebase is more robust and easier to debug.

anselmlong committed:
Merge pull request #1 from anselmlong/branch-A-Assertions

Implement assertions in the codebase

anselmlong committed:
Improve coding quality

There was no coding standard in the project.
I have added a coding standard to the project to
improve the coding quality.

I followed the principles outlined in the Week 5
notes, such as to have a Single Level of Abstraction
and to reduce arrowhead occurences in the code.

This coding standard will be followed for the rest
of the project.

anselmlong committed:
Add streams to TaskList

anselmlong committed:
Merge pull request #2 from anselmlong/branch-A-CodeQuality

Improve coding quality

anselmlong committed:
Merge branch 'master' into branch-A-Streams

anselmlong committed:
Merge pull request #3 from anselmlong/branch-A-Streams

Add streams to TaskList

anselmlong committed:
Create .github folder, add in yml file

anselmlong committed:
Add sort functionality

Various commands to sort the list in different ways
include sort by description, sort by date, sort by type.

anselmlong committed:
Merge pull request #4 from anselmlong/branch-C-Sort

Add sort functionality

anselmlong committed:
Merge branch 'master' of github.com:anselmlong/ip

anselmlong committed:
Update latest version

anselmlong committed:
Improve GUI by changing colors

anselmlong committed:
Merge branch 'branch-A-BetterGui'

anselmlong committed:
Add funny commands to Sigma

anselmlong committed:
Merge branch 'branch-A-Personality'

anselmlong committed:
Add error handling and logging to project

anselmlong committed:
Merge branch 'branch-A-MoreErrorHandling'

anselmlong committed:
Change test cases

anselmlong committed:
Update README.md

anselmlong committed:
Update README.md to include section links

anselmlong committed:
Move README.md to docs
